GitHub Repo Files (Independent Publisher)

Nathalie-Leenders opened the pull request:
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [X ] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [ X] I attest that the connector works and I verified by deploying and testing all the operations. 
- [ X] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [ X] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [ X] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [X ] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [X ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ X] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- ![Githubdatarunsusccesful](https://github.com/microsoft/PowerPlatformConnectors/assets/115070705/d147d31a-8276-48c3-87e8-a5cb569ae4a1)
- [ N/A] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

Nathalie-Leenders committed:
Create readme.md

Nathalie-Leenders committed:
Publishing foldername

Nathalie-Leenders committed:
Create apiProperties.json

Nathalie-Leenders committed:
Create apiDefinition.Swagger.json

Nathalie-Leenders committed:
Update apiProperties.json

Nathalie-Leenders commented:
@microsoft-github-policy-service agree [company="{none}"]

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: microsoft-github-policy-service[bot] ***@***.***>
Sent: Saturday, March 9, 2024 8:53:18 AM
To: microsoft/PowerPlatformConnectors ***@***.***>
Cc: Nathalie Leenders ***@***.***>; Mention ***@***.***>
Subject: Re: [microsoft/PowerPlatformConnectors] Requesting new custom connector - github as a datasource (PR #3290)


@Nathalie-Leenders<https://github.com/Nathalie-Leenders> please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.

@microsoft-github-policy-service agree [company="{your company}"]


Options:

  *   (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.

@microsoft-github-policy-service agree


  *   (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.

@microsoft-github-policy-service agree company="Microsoft"


Contributor License Agreement
Contribution License Agreement

This Contribution License Agreement (“Agreement”) is agreed to by the party signing below (“You”),
and conveys certain license rights to Microsoft Corporation and its affiliates (“Microsoft”) for Your
contributions to Microsoft open source projects. This Agreement is effective as of the latest signature
date below.

  1.  Definitions.
“Code” means the computer software code, whether in human-readable or machine-executable form,
that is delivered by You to Microsoft under this Agreement.
“Project” means any of the projects owned or managed by Microsoft and offered under a license
approved by the Open Source Initiative (www.opensource.org<http://www.opensource.org>).
“Submit” is the act of uploading, submitting, transmitting, or distributing code or other content to any
Project, including but not limited to communication on electronic mailing lists, source code control
systems, and issue tracking systems that are managed by, or on behalf of, the Project for the purpose of
discussing and improving that Project, but excluding communication that is conspicuously marked or
otherwise designated in writing by You as “Not a Submission.”
“Submission” means the Code and any other copyrightable material Submitted by You, including any
associated comments and documentation.
  2.  Your Submission. You must agree to the terms of this Agreement before making a Submission to any
Project. This Agreement covers any and all Submissions that You, now or in the future (except as
described in Section 4 below), Submit to any Project.
  3.  Originality of Work. You represent that each of Your Submissions is entirely Your original work.
Should You wish to Submit materials that are not Your original work, You may Submit them separately
to the Project if You (a) retain all copyright and license information that was in the materials as You
received them, (b) in the description accompanying Your Submission, include the phrase “Submission
containing materials of a third party:” followed by the names of the third party and any licenses or other
restrictions of which You are aware, and (c) follow any other instructions in the Project’s written
guidelines concerning Submissions.
  4.  Your Employer. References to “employer” in this Agreement include Your employer or anyone else
for whom You are acting in making Your Submission, e.g. as a contractor, vendor, or agent. If Your
Submission is made in the course of Your work for an employer or Your employer has intellectual
property rights in Your Submission by contract or applicable law, You must secure permission from Your
employer to make the Submission before signing this Agreement. In that case, the term “You” in this
Agreement will refer to You and the employer collectively. If You change employers in the future and
desire to Submit additional Submissions for the new employer, then You agree to sign a new Agreement
and secure permission from the new employer before Submitting those Submissions.
  5.  Licenses.

  *   Copyright License. You grant Microsoft, and those who receive the Submission directly or
indirectly from Microsoft, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable license in the
Submission to reproduce, prepare derivative works of, publicly display, publicly perform, and distribute
the Submission and such derivative works, and to sublicense any or all of the foregoing rights to third
parties.
  *   Patent License. You grant Microsoft, and those who receive the Submission directly or
indirectly from Microsoft, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable license under
Your patent claims that are necessarily infringed by the Submission or the combination of the
Submission with the Project to which it was Submitted to make, have made, use, offer to sell, sell and
import or otherwise dispose of the Submission alone or with the Project.
  *   Other Rights Reserved. Each party reserves all rights not expressly granted in this Agreement.
No additional licenses or rights whatsoever (including, without limitation, any implied licenses) are
granted by implication, exhaustion, estoppel or otherwise.

  1.  Representations and Warranties. You represent that You are legally entitled to grant the above
licenses. You represent that each of Your Submissions is entirely Your original work (except as You may
have disclosed under Section 3). You represent that You have secured permission from Your employer to
make the Submission in cases where Your Submission is made in the course of Your work for Your
employer or Your employer has intellectual property rights in Your Submission by contract or applicable
law. If You are signing this Agreement on behalf of Your employer, You represent and warrant that You
have the necessary authority to bind the listed employer to the obligations contained in this Agreement.
You are not expected to provide support for Your Submission, unless You choose to do so. UNLESS
REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING, AND EXCEPT FOR THE WARRANTIES
EXPRESSLY STATED IN SECTIONS 3, 4, AND 6, THE SUBMISSION PROVIDED UNDER THIS AGREEMENT IS
PROVIDED WITHOUT WARRANTY OF ANY KIND, INCLUDING, BUT NOT LIMITED TO, ANY WARRANTY OF
NONINFRINGEMENT, MERCHANTABILITY, OR FITNESS FOR A PARTICULAR PURPOSE.
  2.  Notice to Microsoft. You agree to notify Microsoft in writing of any facts or circumstances of which
You later become aware that would make Your representations in this Agreement inaccurate in any
respect.
  3.  Information about Submissions. You agree that contributions to Projects and information about
contributions may be maintained indefinitely and disclosed publicly, including Your name and other
information that You submit with Your Submission.
  4.  Governing Law/Jurisdiction. This Agreement is governed by the laws of the State of Washington, and
the parties consent to exclusive jurisdiction and venue in the federal courts sitting in King County,
Washington, unless no federal subject matter jurisdiction exists, in which case the parties consent to
exclusive jurisdiction and venue in the Superior Court of King County, Washington. The parties waive all
defenses of lack of personal jurisdiction and forum non-conveniens.
  5.  Entire Agreement/Assignment. This Agreement is the entire agreement between the parties, and
supersedes any and all prior agreements, understandings or communications, written or oral, between
the parties relating to the subject matter hereof. This Agreement may be assigned by Microsoft.

—
Reply to this email directly, view it on GitHub<https://github.com/microsoft/PowerPlatformConnectors/pull/3290#issuecomment-1986915903>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/A3N5N4PKGMSKADNPH64NNRTYXM475AVCNFSM6AAAAABEOHSGHWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSOBWHEYTKOJQGM>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

Nathalie-Leenders commented:
@microsoft-github-policy-service agree


Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: microsoft-github-policy-service[bot] ***@***.***>
Sent: Saturday, March 9, 2024 9:07:50 AM
To: microsoft/PowerPlatformConnectors ***@***.***>
Cc: Nathalie Leenders ***@***.***>; Mention ***@***.***>
Subject: Re: [microsoft/PowerPlatformConnectors] Requesting new custom connector - github as a datasource (PR #3290)


@Nathalie-Leenders<https://github.com/Nathalie-Leenders> the command you issued was incorrect. Please try again.

Examples are:

@microsoft-github-policy-service agree


and

@microsoft-github-policy-service agree company="your company"


—
Reply to this email directly, view it on GitHub<https://github.com/microsoft/PowerPlatformConnectors/pull/3290#issuecomment-1986919239>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/A3N5N4PWZXKYIBCECSDJRZLYXM6WNAVCNFSM6AAAAABEOHSGHWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSOBWHEYTSMRTHE>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

vmanoharas reviewed (commented):
Hello @Nathalie-Leenders, 

Can you please go through this document and submit your Independent Publisher Connector?

https://learn.microsoft.com/en-us/connectors/custom-connectors/certification-submission-ip

Nathalie-Leenders commented:
Hi!

I did, what is missing from the request? It's my first time so I probably did something wrong somewhere.

I appreciate any help you can provide.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: vmanoharas ***@***.***>
Sent: Tuesday, March 12, 2024 7:45:18 AM
To: microsoft/PowerPlatformConnectors ***@***.***>
Cc: Nathalie Leenders ***@***.***>; Mention ***@***.***>
Subject: Re: [microsoft/PowerPlatformConnectors] Requesting new custom connector - github as a datasource (PR #3290)


@vmanoharas commented on this pull request.

Hello @Nathalie-Leenders<https://github.com/Nathalie-Leenders>,

Can you please go through this document and submit your Independent Publisher Connector?

https://learn.microsoft.com/en-us/connectors/custom-connectors/certification-submission-ip

—
Reply to this email directly, view it on GitHub<https://github.com/microsoft/PowerPlatformConnectors/pull/3290#pullrequestreview-1931283671>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/A3N5N4NYS75S6KO4AW5RWKTYX4IH5AVCNFSM6AAAAABEOHSGHWVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSMZRGI4DGNRXGE>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

vmanoharas commented on independent-publisher-connectors/Proposal - Github as Datasource/apiProperties.json:
```diff
@@ -0,0 +1,25 @@
+{
+  "properties": {
```
Can you please modify the apiProperties.json file as per the template below. 

https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/Independent%20Publisher/apiProperties.json

vmanoharas commented on independent-publisher-connectors/Proposal - Github as Datasource/readme.md:
```diff
@@ -0,0 +1,38 @@
+## Github as a datassource
+
```
Can you please modify the readme.md file as per the template below?

https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/Independent%20Publisher/readme.md

vmanoharas reviewed (commented):
Hello @Nathalie-Leenders, 

Could you please resolve review comments as well as please follow the schema mentioned in the below link. 

https://github.com/microsoft/PowerPlatformConnectors/tree/dev/templates/Independent%20Publisher

troystaylor commented:
Hi @Nathalie-Leenders, Greetings from the airport.
Let me try to assist with some suggested changes to resolve issues I see in your submitted PR:

- Please change your PR title to Proposal - GitHub Repo Files (Independent Publisher).
- When you believe all issues have been resolved, remove Proposal - from the title.
- In apiProperties.json, please change:
-- Per the February change for [OAuth](https://learn.microsoft.com/en-us/connectors/custom-connectors/#21-oauth-20), please update the redirect Model to GlobalPerConnector. The following line 11 for redirectURL is then unnecessary.
-- Please change the OAuth identity provider to github, as found in the fourth example [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/certification-submission#verify-connection-parameters)
-- Please change stackOwner to GitHub, Inc.
-- Remove the closing curly bracket on the final line and reformat file to correct JSON tab spacing.
- In apiDefinition.swagger.json, please change:
--   For all strings, double quotation marks (") are used, not single quotation marks('). Please change all.
-- There is a mix of YAML and JSON in this file. All JSON object key lines do not start with "- ", e.g. line 36: - "name": ... Also, all JSON object lines need to end with a comma. The pac CLI should have caught those errors, but I would suggest VS Code or an online JSON validator to fix issues.
-- Line 19 for consumes should be an array with the single item application/json. The following produces also needs to be an array.
-- The current line 23 has an unused action path - this must be removed.
-- Your action path is set to require both folder name and folder name 2 - If you want both (or neither?), you could create separate actions. Path parameters are always required based on the defined path.
-- Your path parameters need to include both both x-ms-summary and description. Please review this [doc](https://learn.microsoft.com/en-us/connectors/custom-connectors/openapi-extensions#x-ms-summary).
-- Your response needs to be changed from default to 200. The description can stay the same, and since the response is the markdown file content, you can change the schema to string.
-- At the end of the file, you must also include the [connector metadata](https://learn.microsoft.com/en-us/connectors/custom-connectors/certification-submission#step-4b-product-or-end-service-metadata)

After you make those changes and include them in your branch, please let me know and I can review again.

CC: @ShefaaliP

Nathalie-Leenders committed:
Update apiProperties.json

Made these suggested changes:
-- Per the February change for OAuth, please update the redirect Model to GlobalPerConnector. The following line 11 for redirectURL is then unnecessary.
-- Please change the OAuth identity provider to github, as found in the fourth example here
-- Please change stackOwner to GitHub, Inc.
-- Remove the closing curly bracket on the final line and reformat file to correct JSON tab spacing.

Nathalie-Leenders committed:
Update apiDefinition.Swagger.json

Made changes to the file, hopefully according to Troys suggestions

Nathalie-Leenders committed:
Update apiDefinition.Swagger.json

Nathalie-Leenders commented:
@troystaylor I have made changes to the files, can you please help in checking if they're ok now?

troystaylor commented on independent-publisher-connectors/Proposal - Github as Datasource/apiDefinition.Swagger.json:
```diff
@@ -1,65 +1,104 @@
+    "application/json"
+  ],
+  "paths": {
+    "/companies": {
```
You've changed your action path to /companies, but you should change it back to the original path with the various path parameters. The 200 response I also believe isn't correct for the markdown file string I'm expecting this action returns.

vmanoharas commented:
Hello @Nathalie-Leenders, 

I am following up on this PR, kindly resolve review comments mentioned by Troy.

troystaylor commented:
> Hello @Nathalie-Leenders,
> 
> I am following up on this PR, kindly resolve review comments mentioned by Troy.

@Nathalie-Leenders, I am happy to collaborate on this connector to resolve the comments if you need assistance.

Nathalie-Leenders commented:
Hi all,

Thank you! Due to life getting crazy busy I had to put this one on the backburner, I will definitely retry in getting my manifest ok and reach out to you Troy if I need any help (which I'm pretty sure I will, but will let you know)
Thanks again for following up

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: Troy Taylor ***@***.***>
Sent: Wednesday, June 19, 2024 4:43:37 PM
To: microsoft/PowerPlatformConnectors ***@***.***>
Cc: Nathalie Leenders ***@***.***>; Mention ***@***.***>
Subject: Re: [microsoft/PowerPlatformConnectors] Proposal - GitHub Repo Files (Independent Publisher) (PR #3290)


Hello @Nathalie-Leenders<https://github.com/Nathalie-Leenders>,

I am following up on this PR, kindly resolve review comments mentioned by Troy.

@Nathalie-Leenders<https://github.com/Nathalie-Leenders>, I am happy to collaborate on this connector to resolve the comments if you need assistance.

—
Reply to this email directly, view it on GitHub<https://github.com/microsoft/PowerPlatformConnectors/pull/3290#issuecomment-2178886645>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/A3N5N4PHWXZOPNJ42Q2Y4NLZIGKJTAVCNFSM6AAAAABEOHSGHWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCNZYHA4DMNRUGU>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

Nathalie-Leenders committed:
Merge branch 'microsoft:dev' into dev

Nathalie-Leenders committed:
Update apiProperties.json

Updated per Troy's suggestions:
In apiProperties.json, please change:
-- Per the February change for OAuth, please update the redirect Model to GlobalPerConnector. The following line 11 for redirectURL is then unnecessary.
-- Please change the OAuth identity provider to github, as found in the fourth example here
-- Please change stackOwner to GitHub, Inc.
-- Remove the closing curly bracket on the final line and reformat file to correct JSON tab spacing.

Nathalie-Leenders committed:
Update apiDefinition.Swagger.json

Modified per Troy's suggestions:
In apiDefinition.swagger.json, please change:
-- For all strings, double quotation marks (") are used, not single quotation marks('). Please change all.
-- There is a mix of YAML and JSON in this file. All JSON object key lines do not start with "- ", e.g. line 36: - "name": ... Also, all JSON object lines need to end with a comma. The pac CLI should have caught those errors, but I would suggest VS Code or an online JSON validator to fix issues.
-- Line 19 for consumes should be an array with the single item application/json. The following produces also needs to be an array.
-- The current line 23 has an unused action path - this must be removed.
-- Your action path is set to require both folder name and folder name 2 - If you want both (or neither?), you could create separate actions. Path parameters are always required based on the defined path.
-- Your path parameters need to include both both x-ms-summary and description. Please review this doc.
-- Your response needs to be changed from default to 200. The description can stay the same, and since the response is the markdown file content, you can change the schema to string.
-- At the end of the file, you must also include the connector metadata

Nathalie-Leenders committed:
Update readme.md

Modified per the https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/Independent%20Publisher/readme.md as vmanoharas suggested

Nathalie-Leenders commented:
Hi @troystaylor and @vmanoharas I think I got it (or at least visual studio code copilot helped) I really hope this is OK, can you please check for me? 🙏 I really appreciate the help, this is really not my forte

vmanoharas commented:
> Hi @troystaylor and @vmanoharas I think I got it (or at least visual studio code copilot helped) I really hope this is OK, can you please check for me? 🙏 I really appreciate the help, this is really not my forte

Thank you @Nathalie-Leenders, I will wait @troystaylor to respond, if he has any comments before I review your files.

troystaylor commented:
> Hi @troystaylor and @vmanoharas I think I got it (or at least visual studio code copilot helped) I really hope this is OK, can you please check for me? 🙏 I really appreciate the help, this is really not my forte

Hi Nathalie,
Happy to provide some assistance here. First thing I did was run the paconn CLI tool (or pac works too) against the apiDefinition.swagger.json file: 
![image](https://github.com/microsoft/PowerPlatformConnectors/assets/44444967/32dabd3e-8eb2-4eed-85b3-55ae183791ec)
You've defined one path parameter, but not the other 3-4. As I mentioned previously, path parameters are required, so with the single action you've defined, users will be required to give a folder and subfolder name, which I wouldn't do. The easiest way would be to give two different actions, with one having the single folder and a second action to put the file in a subfolder. I've also thought about this and you could use custom code if you set the folder name parameters to query parameters (which don't have to be required). You also need to define Reponame and Markdownfile - those are required.
The second major error is the summary length, it cannot be over 80 characters. I would suggest 'Get GitHub file' and 'Get GitHub file in subfolder', fixing the additional error that summaries cannot end in punctuation.
Next, the connector metadata property 'Privacy policy' is missing.
The final error is the missing key for each path parameters - "x-ms-url-encoding": "single"
The CLI tool doesn't catch it, but GitHub should have the correct capitalization for every instance. Another suggestion I would use is to use either camelcase or PascalCase for all your parameter - it makes the code easier to read. I was also thinking about the usefulness of the connector and while I think being able to directly return markdown as a string to Power Automate is useful, I think being able to retrieve any file raw is more useful, so I've changed the produces to application/vnd.github.raw+json.
In looking at your apiProperties file, I don't understand what providing the OAuth setting as part of the URL path is suppose to do. The more I thought about with the host you are using, I realized that you would only be able to retrieve public files, because the OAuth would never be valid against that host - the OAuth settings need to be removed. To make this connector work with OAuth, it needs to properly call the api.github.com host and use the correct endpoint: https://docs.github.com/en/rest/repos/contents?apiVersion=2022-11-28#get-repository-content . I'm going to work on the actions using OAuth and I'll share my suggestions soon.
@vmanoharas the current artifacts are not ready for certification.

troystaylor commented:
Hi Nathalie,
Please review the files I've created that have passed CLI validation: https://github.com/troystaylor/Testing/tree/main/GitHub%20Data
As I mentioned, this action will work will any public repo. I am going to continue to investigate the endpoints for the Repositories API and will submit them for a separate connector.

Nathalie-Leenders committed:
Merge branch 'microsoft:dev' into dev

Nathalie-Leenders committed:
Updating swagger file

Nathalie-Leenders commented:
I've modified my file, can you please review at your convenience? My past 6 months were a bit crazy but looking forward to getting this published. @vmanoharas

troystaylor commented:
@Nathalie-Leenders Hi Nathalie, sorry for the delay - I was trying to stay offline during the holidays. 

- You are missing a "securityDefinitions" object, which for this connector can be blank. 
- Your "schemes", "consumes" and "produces" arrays need to be formatted as well so that the array values are new lines.
- Your "summary" value for the get needs to be shorter than 80 characters. You have used the same values for summary and description, so I would suggest just "Get file"
- All path parameters are missing key "x-ms-url-encoding" with value "single"
- You currently have both Foldername and Foldername2 as required, which will cause users to not be able to download files in the root of the repo - I would suggest combining Foldername, Foldername2 and Markdownfile into one parameter called "Filewithpath", which would allow users to input the path included forward slashes. Also, all files, not just markdown files, can be downloaded as strings.

![image](https://github.com/user-attachments/assets/74c394d9-9591-4794-8914-027d92d996e9)
![image](https://github.com/user-attachments/assets/09a22fa1-9806-4ba0-b455-9681a4de0acc)

I would also like to suggest before you update the Swagger file again, please try validating with either the pac or paconn CLI tools - this will give you a list of errors and warning:

![image](https://github.com/user-attachments/assets/274def1c-8302-4fdc-a45c-025843bc1089)

Nathalie-Leenders committed:
pushing changes, it should be good now

Nathalie-Leenders commented:
I have just pushed through final updates, CLI clears it, it should be ok now. Thanks for your tips @troystaylor  greatly appreciated

Nathalie-Leenders commented:
@vmanoharas @troystaylor Anything you need from me? We should be there 🙏

vmanoharas commented:
Hi @troystaylor, 

Could you please review that your changes are addressed before I proceed for the review?

troystaylor commented:
I can confirm changes now work.
@Nathalie-Leenders - please correct the spelling so that GitHub has uppercase G and H in all instances. Then remove Proposal from the PR title and @vmanoharas can start the review.

Nathalie-Leenders committed:
Updating github to GitHub

Nathalie-Leenders commented:
@troystaylor I have changed all Github to GitHub, and changed the name of the PR. @vmanoharas Can you please start the review?
Thank you both so much for your help.

vmanoharas commented on independent-publisher-connectors/Proposal - Github as Datasource/readme.md:
```diff
@@ -0,0 +1,48 @@
+- GitHub name is your user account for GitHub, or from the account you're trying to pull data from.
+- Repo name is the name of your repo.
+- Foldername is the name of the first folder
+- Foldername2 is the underlying folders. If there are more than use use foldername/foldername2/foldername3 in this parameter if needed.
```
Please remove repeated word "use use"

Nathalie-Leenders replied:
I have updated and removed use use

vmanoharas reviewed (commented):
Thank you @troystaylor for your help on this. 

Thank @Nathalie-Leenders for resolving all the comments.  Swagger validation passed successfully, please resolve one comment.

Nathalie-Leenders committed:
Update readme.md

Removed use use

vmanoharas reviewed (dismissed):
Dear Partner

I hope you are doing well.
Congratulations, your pull request is approved. We will move forward with the certification process. We are preparing your connector for production deployment and will queue up for the next deployment schedule. Once your connector onboards the next deployment schedule, it starts to deploy your connector in our production environments that typically takes 3-4 weeks.
Please let us know if you have any questions.
Thank you very much for working with us.

Nathalie-Leenders commented:
@vmanoharas Do you know when it will be published? Waiting patiently 🙏

vmanoharas commented:
Hi @Nathalie-Leenders,  We have some challenge to certify oAuth IP connectors, we are working with security team.  Please allow us some more time.

Nathalie-Leenders commented:
Is there anything I can do to help? My connector has no authentication in it, it's just a get command to the github repo

vmanoharas commented:
> Hi @Nathalie-Leenders, We have some challenge to certify oAuth IP connectors, we are working with security team. Please allow us some more time.



> Is there anything I can do to help? My connector has no authentication in it, it's just a get command to the github repo

Hi @Nathalie-Leenders, 

Your connector is oAuth enabled connector, we are working with our security team that how should we move the oAuth IP connectors.  Please allow us sometime.  However, no action pending from your end.

Nathalie-Leenders commented:
We're now 3 months further, any update? @vmanoharas

vmanoharas commented:
Hi @Nathalie-Leenders, 

Apologies for the delay. 

Thank you for your continued partnership and collaboration.
To uphold the highest security standards across our ecosystem, we will be pausing support for OAuth connectors moving forward.
We kindly ask that you update your connector to use either Basic or API Key authentication and resubmit your connector artifacts accordingly.

We truly appreciate your understanding and support as we continue to strengthen secure integration practices together.

Additionally, please follow the below documentation and upload connector package zip file.

https://learn.microsoft.com/en-us/connectors/custom-connectors/certification-submission-ip#step-6-go-through-connector-validation-and-certification-process

Please refer this PR for example. In this PR troystaylor uploaded zip file, please upload only zip file not other files.

https://github.com/microsoft/PowerPlatformConnectors/pull/3908/files

Nathalie-Leenders committed:
Remove GitHub OAuth connection parameters

Removed connection parameters for GitHub OAuth settings.

Nathalie-Leenders commented:
@troystaylor  @vmanoharas  it should be good nlw, thank you. Its an open call to a public location, no oath.

Nathalie-Leenders committed:
Fix JSON formatting in apiProperties.json

vmanoharas commented:
Hi @Nathalie-Leenders, 

Our connector certification process has been changed.  Kindly refer below links and upload packages and follow [folder structure](https://learn.microsoft.com/en-us/connectors/custom-connectors/certification-submission#package-your-connector-files). 

https://learn.microsoft.com/en-us/connectors/custom-connectors/certification-submission
https://learn.microsoft.com/en-us/connectors/custom-connectors/certification-submission-ip

Additionally, we have sent a [onevet](https://learn.microsoft.com/en-us/connectors/custom-connectors/certification-submission-ip#step-5-onevet-verification) request.  Please fille the form.

Nathalie-Leenders commented:
@vmanoharas I can appreciate processes to be changed, however in another connector that was also done earlier with the old structure I see it was approved a few days ago. I have tried to change to the new structure and it's unclear in the documentation which steps I need to follow or change.

@troystaylor could you please help?

I'm really hoping we can get this approved as soon as possible, I originally submitted this March 2024.

troystaylor commented:
@Nathalie-Leenders Please download this file and then upload to your repo folder where the other connector files are:
https://github.com/troystaylor/Testing/blob/main/ConnectorPackage.zip

Nathalie-Leenders committed:
Merge branch 'microsoft:dev' into dev

Nathalie-Leenders committed:
Adding connector package

Nathalie-Leenders commented:
@troystaylor Thank you very much! I have added it.

@vmanoharas Can you please check now?

vmanoharas commented:
Thank you @troystaylor here for the assistance. 

@Nathalie-Leenders, we have sent you the onevet request form.  Please fill the form and notify here.  So that we can proceed for the next steps.

Nathalie-Leenders commented:
Hi! @vmanoharas  I have already filled it out 2 days ago :)

Nathalie-Leenders commented:
I got another one so just filled that one in as well

vmanoharas commented:
Thanks for filling the form @Nathalie-Leenders. 

We have shared Onevet link, please follow the steps mentioned in the below link and complete the process. 

https://learn.microsoft.com/en-us/connectors/custom-connectors/certification-submission-ip#step-5-onevet-verification

Nathalie-Leenders commented:
Hi @vmanoharas I have verified and it should be good now, thank you for your help in this

vmanoharas commented:
certify-connector

certificationteam commented:
Congratulations — your certification request has been approved. Your connector will be included in the upcoming deployment schedule for production release and the supporting documentation will be published. This end-to-end process typically takes 3–4 weeks.

Many Thanks,
Copilot Connector Certification Team

vmanoharas commented:
Hello Partner,
I hope you are doing well.
Congratulations, the deployment of your connector and documentation page was already completed, 
Thank you very much for working with us.
